### PR TITLE
build(manager/electron): 👷 add option to use GCP KMS for signing

### PR DESCRIPTION
**Pull request by jyyi1**
In this changeset:

- upgraded `jsign.jar` to version `4.2`
- add an option to leverage `gcp-hsm` certificate to sign Windows executables
- please note that [GCP does not support SHA-1](https://cloud.google.com/kms/docs/algorithms#asymmetric_signing_algorithms), and here are the consequences of not adding SHA-1 to the binary signatures:
   1. [SHA-1 is deprecated by Microsoft](https://techcommunity.microsoft.com/t5/windows-it-pro-blog/sha-1-windows-content-to-be-retired-august-3-2020/ba-p/1544373)
   2. The only affected OSes are Windows 7 and below ([support ended already](https://www.microsoft.com/en-us/windows/end-of-support)). Windows 7 (6.1.7600) users can still upgrade to Windows 7 SP1 (6.1.7601) which supports SHA-2.
   3. Users can still run our app on old Windows systems (Windows 7), despite that the following warning dialog might show up:
image

### CODE REVIEW

File: src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs [hunk 1]
```diff
@@ -23,6 +23,13 @@
 async function electronBuilderEntryPoint(configuration) {
   // CommonJS module is required, ES6 module is not supported by electron-builder
   const {signWindowsExecutable} = await import('./sign_windows_executable.mjs');
+  if (configuration.hash === 'sha1') {
+    // SHA-1 signatures are retired and should be replaced by SHA-2, Windows 7
+    // users are required to upgrade to Windows 7 SP1 (or install the standalone
+    // update KB4474419)
+    console.info('skipping SHA-1 signature');
```
Comment (fortuna) [inline src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs:30]: We should make this fail instead.
sha1 should be removed from the configuration, so it's clear what we are doing. Silent failures are dangerous.

**Review by fortuna — CHANGES_REQUESTED** [review]

File: src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs (hunk 1 above)
Comment (jyyi1) [reply src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs:30]: Agree to disable it to explicit configuration, `electron-builder` does not provide a way to remove SHA-1: ["Windows is dual code-signed (SHA1 & SHA256 hashing algorithms)."](https://www.electron.build/code-signing.html). I think we can introduce an environment variable to explicitly disable the dual signing behavior (e.g., `WINDOWS_SIGNING_ELECTRON_BUILDER_DISABLE_DUAL_SIGNING`).

Comment (fortuna) [reply src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs:30]: See `signingHashAlgorithms` from https://www.electron.build/configuration/win.html

Comment (jyyi1) [reply src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs:30]: Great, but I would like to have two different hash algorithms for different signing methods, for example, having [`sha-1`, `sha-256`] for USB stick while having [`sha-256`] only for GCP HSM.

Comment (jyyi1) [reply src/server_manager/electron_app/windows/electron_builder_signing_plugin.cjs]: I deleted these code and still keep the dual-signing option, if people want to use GCP, they need to explicitly disable sha-1 in the `electron-builder` configuration.

**Review by fortuna — APPROVED** [review]

**Pull request merged**